openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java

```diff
@@ -394,34 +395,41 @@ public ResultList<T> listAfter(UriInfo uriInfo, Fields fields, ListFilter filter
   @Transaction
   public ResultList<T> listAfterWithSkipFailure(
       UriInfo uriInfo, Fields fields, ListFilter filter, int limitParam, String after) throws IOException {
-    List<String> errors = new ArrayList<>();
+    Map<UUID, String> errors = new LinkedHashMap<>();
+    Map<UUID, T> entities = new LinkedHashMap<>();
     int total = dao.listCount(filter);
-    List<T> entities = new ArrayList<>();
     if (limitParam > 0) {
       // forward scrolling, if after == null then first page is being asked
       List<String> jsons = dao.listAfter(filter, limitParam + 1, after == null ? "" : RestUtil.decodeCursor(after));
 
       for (String json : jsons) {
         try {
           T entity = withHref(uriInfo, setFieldsInternal(JsonUtils.readValue(json, entityClass), fields));
-          entities.add(entity);
+          entities.put(entity.getId(), entity);
         } catch (Exception e) {
           LOG.error("Failed in Set Fields for Entity with Json : {}", json);
-          errors.add(json);
+          errors.put(JsonUtils.readValue(json, entityClass).getId(), json);
         }
       }
 
       String beforeCursor;
       String afterCursor = null;
-      beforeCursor = after == null ? null : entities.get(0).getFullyQualifiedName();
-      if (entities.size() > limitParam) { // If extra result exists, then next page exists - return after cursor
-        entities.remove(limitParam);
-        afterCursor = entities.get(limitParam - 1).getFullyQualifiedName();
+      beforeCursor = after == null ? null : JsonUtils.readValue(jsons.get(0), entityClass).getFullyQualifiedName();
```

Comment:
@mohityadav766 not sure what we are doing with this. Ideally there should never be an error in doing pagination. We need to identify those bugs, but it looks like we are trying to delete them?

Reply:
@harshach this is the same logic that we had, just that we look for the data fetched in list of jsons till limit param, this if the error is at limitParam + 1 we remove it since in the next fetch it is fetched again , resulting the errored entity 2 be accounted 2 times